src/strategies.rs

```diff
@@ -0,0 +1,154 @@
+use crate::config::DurationIterator;
+use rand::{rngs::StdRng, Rng, SeedableRng};
+use std::time::Duration;
+
+pub(crate) fn get_standard_reconnect_strategy() -> DurationIterator {
+    let initial_attempts = vec![
+        Duration::from_secs(5),
+        Duration::from_secs(10),
+        Duration::from_secs(20),
+        Duration::from_secs(30),
+        Duration::from_secs(40),
+        Duration::from_secs(50),
+        Duration::from_secs(60),
+        Duration::from_secs(60 * 2),
+        Duration::from_secs(60 * 5),
+        Duration::from_secs(60 * 10),
+        Duration::from_secs(60 * 20),
+    ];
+
+    let repeat = std::iter::repeat(Duration::from_secs(60 * 30));
+
+    let forever_iterator = initial_attempts.into_iter().chain(repeat);
+
+    Box::new(forever_iterator)
+}
+
+/// Type used for defining the exponential backoff strategy.
+/// # Examples
+///
+/// ```
+/// use std::time::Duration;
+/// use stubborn_io::{ReconnectOptions, ExpBackoffStrategy};
+///
+/// // With the below strategy, the stubborn-io item will try to reconnect infinitely,
+/// // waiting an exponentially increasing (by 2) value with 5% random jitter. Once the
+/// // wait would otherwise exceed the maxiumum of 30 seconds, it will instead wait 30
+/// // seconds.
+///
+/// let options = ReconnectOptions::new().with_retries_generator(|| {
+///     ExpBackoffStrategy::new(Duration::from_secs(1), 2, 0.05)
+///         .with_max(Duration::from_secs(30))
+/// });
+/// ```
+pub struct ExpBackoffStrategy {
+    min: Duration,
+    max: Option<Duration>,
+    factor: u32,
+    jitter: f64,
+    seed: Option<u64>,
+}
+
+impl ExpBackoffStrategy {
```

Comment:
I wonder if we should make this strategy the default one entirely, with sensible default values